digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -617,6 +619,57 @@ trait LaneOperations {
     }
   }
 
+  def updateTrafficDirectionChange(linkProperty: LinkProperties, username: Option[String]): Unit = {
+    val allLanes = dao.fetchAllLanesByLinkIds(Seq(linkProperty.linkId), includeExpired = true)
+    val lanesInUse = allLanes.filter(lane => lane.expired == false)
+    val sideCodeToPreserve = toSideCode(linkProperty.trafficDirection).value
+
+    if (lanesInUse.size == 2) {
+      if (linkProperty.trafficDirection.isOneWay) {
+        val lanesToExpire = lanesInUse.filter(lane => lane.sideCode != sideCodeToPreserve)
+        if (lanesToExpire.size == 2) {
+          processLaneChanges(lanesToExpire.head, sideCodeToPreserve, lanesToExpire.head.laneCode)
+          processLaneChanges(lanesToExpire.last, sideCodeToPreserve, lanesToExpire.last.laneCode)
+        } else if (lanesToExpire.size == 1) {
+          processLaneChanges(lanesToExpire.head, sideCodeToPreserve,2)
+        }
+      } else {
+        val laneToExpire = lanesInUse.find(lane => lane.laneCode != 1)
+        laneToExpire match {
+          case Some(lane) =>
+            val newSideCode = switch(SideCode.apply(lane.sideCode)).value
+            processLaneChanges(lane, newSideCode, 1)
+          case None =>
+            //do nothing
+        }
+      }
+
+    } else if (lanesInUse.size == 1) {
+      val laneToExpire = lanesInUse.head
+      if (laneToExpire.sideCode != sideCodeToPreserve) {
+        processLaneChanges(laneToExpire, sideCodeToPreserve, laneToExpire.laneCode)
+      }
+    }
+
+    def processLaneChanges(laneToExpire: PersistedLane, newSideCode: Int, newLaneCode: Int) = {
```

Comment:
```suggestion
    def processLaneChanges(laneToExpire: PersistedLane, newSideCode: SideCode, newLaneCode: Int) = {
```
newLaneCode vois olla tyyppiä LaneNumberOneDigit mut ei ehkä oo tarpeen

Reply:
Hmm, PersistedLane-luokassa sidecode on kuitenkin tyyppiä Int eikä toi Sidecode enum?

Reply:
siis ajatuksena on piilottaa mystiset numerot piiloon ja sitten käyttää niitä numeroita kun on tarvetta.
jolloin toi ottaisi sisään SideCode.BothDirections vaikkapa ja se sitten muunetaan numeroksi  BothDirections.value kun mennään dao luokkaan